### DISCUSSION TITLE
Changing the value and CSS of the slider from external source

### DISCUSSION THREAD

**Discussion by atirim-jcrew** [radix-ui/primitives#1954] [Help] [answered]
Hello, 

I am build a sort of theme customizer and have 3 sliders ( hue, saturation, lightness). I also have a function that generates shades. 

<img width="1007" alt="image" src="https://user-images.githubusercontent.com/103128911/218668382-5528f931-1f2b-4acf-86c2-d2ee62c5c2f2.png">

It looks like this. When I click on a shade, it changes the values of the slider. The value changes, but the actual calculations on the range and thumb do not change. Is there a workaround I am missing or some custom implementation should be required ?

**Comment by joaom00** [comment, marked as answer]
How are you change the slider value?

**Comment by atirim-jcrew** [comment]
I change it with the same on change function as I do the slider .

```
In Slider : 
 onValueChange: (e) => setLightness({ lightness: e[0], color: type }),

In Color Pallete: 
I have a function that changes all 3 at once: 

  const setColor = (h, s, l, type) => {
    setHue({ hue: h, color: type });
    setSaturation({ saturation: s, color: type });
    setLightness({ lightness: l, color: type });
  };


```

**Reply by joaom00** [reply]
Can you provide a minimal reproduction on a sandbox?

**Reply by atirim-jcrew** [reply]
I'll do this now.

**Reply by atirim-jcrew** [reply]
https://codesandbox.io/s/tailwind-css-and-react-forked-q5rukh?file=/src/App.js here it is. Thank you

**Reply by joaom00** [reply]
in the `Slide` component you are not passing `value` prop to `Root` part and the value needs to be an array, so `value={hue} => value={[hue]}`.

Here the working example https://codesandbox.io/s/tailwind-css-and-react-forked-2neeqg

**Reply by atirim-jcrew** [reply]
That makes sense, thank you so much for taking the time to help. Really appreciate it.

**Comment by atirim-jcrew** [comment]
Also, the value does update accordingly, but the CSS part of it, does not.

**Reply by benoitgrelard** [reply]
What do you mean? @joaom00 was spot on pointing out the mistake in his updated sandbox. That seems to work all fine there.

**Reply by atirim-jcrew** [reply]
Yes, he was, and I said thank you to him for pointing out my mistake. I don't see where is the problem?

**Reply by benoitgrelard** [reply]
ha ok, perhaps it was lost in translation.

> Also, the value does update accordingly, but the CSS part of it, does not.

I thought you were asking another question here.
No worries!

**Reply by andyluak** [reply]
No, instead of editing the original question I replied. I get the confusion now.
